diff --git a/builds/posix/Makefile.in b/builds/posix/Makefile.in
index 063d2abedc..8f698b0a2a 100644
--- a/builds/posix/Makefile.in
+++ b/builds/posix/Makefile.in
@@ -197,7 +197,7 @@ ifeq ($(CLIENT_ONLY_FLG),N)
 	CC="$(CC)" CFLAGS="$(CFLAGS)" AR="$(AR)" $(MAKE) -C $(ROOT)/extern/btyacc
 endif
 
-	CXX="$(CXX)" CXXFLAGS="$(CXXFLAGS)" $(MAKE) -C $(ROOT)/extern/cloop TARGET=release WITH_FPC=0 BUILD_DIR=$(TMP_ROOT)/cloop OUT_DIR=$(GEN_ROOT)/$(TARGET)/cloop core
+	CXX="$(CXX)" $(MAKE) -C $(ROOT)/extern/cloop TARGET=release WITH_FPC=0 BUILD_DIR=$(TMP_ROOT)/cloop OUT_DIR=$(GEN_ROOT)/$(TARGET)/cloop core
 
 	CC="$(CC)" CFLAGS="$(CFLAGS)" AR="$(AR)" $(MAKE) -C $(ROOT)/extern/decNumber
 	ln -sf $(ROOT)/extern/decNumber/libdecFloat.a $(STATIC_LIB)
diff --git a/builds/posix/darwin.defaults b/builds/posix/darwin.defaults
index 8012af3ea7..d8350092c2 100644
--- a/builds/posix/darwin.defaults
+++ b/builds/posix/darwin.defaults
@@ -21,7 +21,7 @@
 #FirebirdInstallPrefix=
 LIB_BUNDLE_OPTIONS=$(LD_FLAGS) -bundle
 LIB_LINK_OPTIONS=$(LD_FLAGS) -dynamiclib
-LIB_LINK_SONAME:=-current_version @FIREBIRD_VERSION@ -compatibility_version @FIREBIRD_VERSION@ -seg1addr 0x30000000
+LIB_LINK_SONAME:=-current_version @FIREBIRD_VERSION@ -compatibility_version @FIREBIRD_VERSION@
 LIB_LINK_MAPFILE=-Wl,-exported_symbols_list,$(1)
 
 ADD_LIBS:=-lobjc -framework Foundation -framework Security
diff --git a/builds/posix/make.defaults b/builds/posix/make.defaults
index 2bbf905236..0d5717dde4 100755
--- a/builds/posix/make.defaults
+++ b/builds/posix/make.defaults
@@ -365,41 +365,41 @@ LIB_LINK_MAPFILE= -Wl,--version-script,$(1)
 FIREBIRD_LIBRARY_LINK= -L$(LIB) -L$(STATIC_LIB) -lfbclient $(MATHLIB) $(CRYPTLIB)
 
 EXE_LINK_OPTIONS= $(LDFLAGS) $(THR_FLAGS) $(UNDEF_FLAGS) $(LIB_PATH_OPTS) $(call LINK_DARWIN_RPATH,..) $(LINK_EMPTY_SYMBOLS)
-LIB_LINK_OPTIONS= $(LDFLAGS) $(THR_FLAGS) -shared
+LIB_LINK_OPTIONS= $(LDFLAGS) $(THR_FLAGS) -shared $(call LINK_DARWIN_RPATH,..)
 
 FB_DAEMON = $(BIN)/firebird$(EXEC_EXT)
 
 # Per-library link rules
 LINK_UDF = $(LIB_LINK) $(LIB_LINK_OPTIONS) $(call LIB_LINK_SONAME,$(1).$(SHRLIB_EXT)) $(UNDEF_FLAGS)\
- $(call LIB_LINK_RPATH,lib) $(call LINK_DARWIN_RPATH,..)
+ $(call LIB_LINK_RPATH,lib)
 LINK_UDF_LIBS = $(THR_LIBS) -L$(LIB) -L$(STATIC_LIB) -lib_util $(SO_LINK_LIBS)
 
 LINK_IB_UTIL = $(LIB_LINK) $(LINK_IBUTIL_SYMBOLS) $(LIB_LINK_OPTIONS) $(UNDEF_FLAGS)\
- $(call LIB_LINK_SONAME,$(IbUtilLibraryName)) $(call LIB_LINK_RPATH,lib) $(call LINK_DARWIN_RPATH,..)
+ $(call LIB_LINK_SONAME,$(IbUtilLibraryName)) $(call LIB_LINK_RPATH,lib)
 LINK_IB_UTIL_LIBS = $(THR_LIBS)
 
 LINK_INTL = $(LIB_LINK) $(LINK_FBINTL_SYMBOLS) $(LIB_LINK_OPTIONS) $(UNDEF_FLAGS)\
- $(call LIB_LINK_SONAME,libfbintl.$(SHRLIB_EXT).1) $(call LIB_LINK_RPATH,lib) $(call LINK_DARWIN_RPATH,..)
+ $(call LIB_LINK_SONAME,libfbintl.$(SHRLIB_EXT).1) $(call LIB_LINK_RPATH,lib)
 LINK_INTL_LIBS = -L$(LIB) -L$(STATIC_LIB) $(SO_LINK_LIBS) $(FIREBIRD_LIBRARY_LINK)
 
 LINK_TRACE = $(LIB_LINK) $(LINK_PLUGIN_SYMBOLS) $(LIB_LINK_OPTIONS) $(UNDEF_FLAGS)\
- $(call LIB_LINK_SONAME,$(LIB_PREFIX)fbtrace.$(SHRLIB_EXT).0) $(call LIB_LINK_RPATH,lib) $(call LINK_DARWIN_RPATH,..)
+ $(call LIB_LINK_SONAME,$(LIB_PREFIX)fbtrace.$(SHRLIB_EXT).0) $(call LIB_LINK_RPATH,lib)
 LINK_TRACE_LIBS = -L$(LIB) -L$(STATIC_LIB) $(SO_LINK_LIBS)
 
 LINK_FIREBIRD = $(LIB_LINK) $(LINK_FIREBIRD_SYMBOLS) $(LIB_LINK_OPTIONS) $(LIB_FIREBIRD_OPTIONS) $(UNDEF_FLAGS)\
- $(call LIB_LINK_SONAME,$(LibrarySoName)) $(call LIB_LINK_RPATH,lib) $(call LINK_DARWIN_RPATH,..)
+ $(call LIB_LINK_SONAME,$(LibrarySoName)) $(call LIB_LINK_RPATH,lib)
 LINK_FIREBIRD_LIBS = -L$(LIB) -L$(STATIC_LIB) $(LIB_GUI) $(SO_LINK_LIBS) $(MATHLIB)
 
 LINK_ENGINE = $(LIB_LINK) $(LINK_PLUGIN_SYMBOLS) $(LIB_LINK_OPTIONS) $(LIB_FIREBIRD_OPTIONS) $(UNDEF_FLAGS)\
- $(call LIB_LINK_SONAME,$(EngineSoName)) $(call LIB_LINK_RPATH,lib) $(call LINK_DARWIN_RPATH,..)
+ $(call LIB_LINK_SONAME,$(EngineSoName)) $(call LIB_LINK_RPATH,lib)
 LINK_ENGINE_LIBS = $(LINK_FIREBIRD_LIBS) $(RE2LIB) $(LIBCDSLIB) $(FIREBIRD_LIBRARY_LINK)
 
-LINK_UDRENG = $(LIB_LINK) $(LINK_PLUGIN_SYMBOLS) $(LIB_LINK_OPTIONS) $(call LIB_LINK_RPATH,lib) $(UNDEF_FLAGS)\
- $(call LINK_DARWIN_RPATH,..)
+LINK_UDRENG = $(LIB_LINK) $(LINK_PLUGIN_SYMBOLS) $(LIB_LINK_OPTIONS) $(call LIB_LINK_RPATH,lib) $(UNDEF_FLAGS)
+
 LINK_UDRENG_LIBS = -L$(LIB) -L$(STATIC_LIB) $(SO_LINK_LIBS)
 
-LINK_PLUGIN = $(LIB_LINK) $(LINK_PLUGIN_SYMBOLS) $(LIB_LINK_OPTIONS) $(call LIB_LINK_RPATH,lib) $(UNDEF_FLAGS)\
- $(call LINK_DARWIN_RPATH,..)
+LINK_PLUGIN = $(LIB_LINK) $(LINK_PLUGIN_SYMBOLS) $(LIB_LINK_OPTIONS) $(call LIB_LINK_RPATH,lib) $(UNDEF_FLAGS)
+
 LINK_PLUG_LIBS = -L$(LIB) -L$(STATIC_LIB) $(SO_LINK_LIBS)
 
 # Pay attention - we place common library into obj, not lib dir
diff --git a/builds/posix/prefix.darwin_aarch64 b/builds/posix/prefix.darwin_aarch64
index 7f2da73ccb..c179da7fc3 100644
--- a/builds/posix/prefix.darwin_aarch64
+++ b/builds/posix/prefix.darwin_aarch64
@@ -33,17 +33,19 @@
 #DYLD_PRINT_LIBRARIES=1
 #export DYLD_PRINT_LIBRARIES
 
-MACOSX_DEPLOYMENT_TARGET=11.0
+MACOSX_DEPLOYMENT_TARGET=12.7
 export MACOSX_DEPLOYMENT_TARGET
 
-PROD_FLAGS=-DDARWIN -DARM64 -pipe -O2 -MMD -fPIC -fno-common -mmacosx-version-min=11.0
-DEV_FLAGS=-ggdb -DDARWIN -DARM64 -pipe -MMD -fPIC -fno-omit-frame-pointer -fno-common -Wall -fno-optimize-sibling-calls -mmacosx-version-min=11.0 -Wno-non-virtual-dtor
+PLATFORM_PLUSPLUS_FLAGS=-Wno-invalid-offsetof
+
+PROD_FLAGS=-DDARWIN -DARM64 -pipe -O2 -MMD -fPIC -fno-common -mmacosx-version-min=12.7
+DEV_FLAGS=-ggdb -DDARWIN -DARM64 -pipe -MMD -fPIC -fno-omit-frame-pointer -fno-common -Wall -fno-optimize-sibling-calls -mmacosx-version-min=12.7 -Wno-non-virtual-dtor
 CXXFLAGS:=$(CXXFLAGS) -fvisibility-inlines-hidden -fvisibility=hidden
 
 UNDEF_PLATFORM=
 
-LINK_LIBS+=-liconv
+#LINK_LIBS+=-liconv
 #MATHLIB=$(ROOT)/extern/libtommath/.libs/libtommath.a
-SO_LINK_LIBS+=-liconv
+#SO_LINK_LIBS+=-liconv
 
 include $(ROOT)/gen/darwin.defaults
diff --git a/builds/posix/prefix.darwin_x86_64 b/builds/posix/prefix.darwin_x86_64
index e4640893da..71790506b2 100644
--- a/builds/posix/prefix.darwin_x86_64
+++ b/builds/posix/prefix.darwin_x86_64
@@ -28,15 +28,17 @@ export DYLD_LIBRARY_PATH
 DYLD_FALLBACK_LIBRARY_PATH=/opt/local/lib
 export DYLD_FALLBACK_LIBRARY_PATH
 
-MACOSX_DEPLOYMENT_TARGET=10.9
+MACOSX_DEPLOYMENT_TARGET=12.7
 export MACOSX_DEPLOYMENT_TARGET
 
-PROD_FLAGS=-O1 -DDARWIN -pipe -MMD -fPIC -fno-common -mmacosx-version-min=10.7
-DEV_FLAGS=-ggdb -DDARWIN -pipe -MMD -fPIC -fno-omit-frame-pointer -fno-common -Werror=delete-incomplete -Wall -fno-optimize-sibling-calls -mmacosx-version-min=10.7 -Wno-non-virtual-dtor
+PLATFORM_PLUSPLUS_FLAGS=-Wno-invalid-offsetof
+
+PROD_FLAGS=-O1 -DDARWIN -pipe -MMD -fPIC -fno-common -mmacosx-version-min=12.7
+DEV_FLAGS=-ggdb -DDARWIN -pipe -MMD -fPIC -fno-omit-frame-pointer -fno-common -Wall -fno-optimize-sibling-calls -mmacosx-version-min=12.7 -Wno-non-virtual-dtor
 CXXFLAGS:=$(CXXFLAGS) -fvisibility-inlines-hidden -fvisibility=hidden -msse4
 
-LD_FLAGS+=-liconv
-FIREBIRD_LIBRARY_LINK+=-liconv
+#LD_FLAGS+=-liconv
+#FIREBIRD_LIBRARY_LINK+=-liconv
 UNDEF_PLATFORM=
 
 include $(ROOT)/gen/darwin.defaults
diff --git a/configure.ac b/configure.ac
index 7f575754e2..32adf0ac86 100644
--- a/configure.ac
+++ b/configure.ac
@@ -715,7 +715,7 @@ case "$build" in
   *-*-darwin*)
     CFLAGS="$CFLAGS -I${VCPKG_INSTALLED}/include"
     CXXFLAGS="$CXXFLAGS -nostdinc++ -isystem ${VCPKG_INSTALLED}/include/c++/v1 -I${VCPKG_INSTALLED}/include"
-    LDFLAGS="$LDFLAGS -nostdlib++ -stdlib=libc++ -L${VCPKG_INSTALLED}/lib -lc++"
+    LDFLAGS="$LDFLAGS -nostdlib++ -L${VCPKG_INSTALLED}/lib -lc++"
     ;;
 esac
 
diff --git a/extern/cloop/Makefile b/extern/cloop/Makefile
index 2613b88a5a..ec7d67594e 100644
--- a/extern/cloop/Makefile
+++ b/extern/cloop/Makefile
@@ -41,10 +41,6 @@ ifeq ($(TARGET),release)
 	CXX_FLAGS += -O3
 endif
 
-ifeq ($(shell uname),Darwin)
-CXX_FLAGS += -stdlib=libc++
-endif
-
 ifeq ($(TARGET),debug)
 	FPC_FLAGS += -g
 	LD_FLAGS += -ggdb
diff --git a/extern/libtomcrypt/makefile.shared b/extern/libtomcrypt/makefile.shared
index b4e7c43062..c0f17a4716 100644
--- a/extern/libtomcrypt/makefile.shared
+++ b/extern/libtomcrypt/makefile.shared
@@ -43,10 +43,10 @@ include makefile_include.mk
 
 #ciphers come in two flavours... enc+dec and enc
 src/ciphers/aes/aes_enc.o: src/ciphers/aes/aes.c src/ciphers/aes/aes_tab.c
-	$(LTCOMPILE) $(LTC_CFLAGS) $(CPPFLAGS) $(LTC_LDFLAGS) -DENCRYPT_ONLY -c src/ciphers/aes/aes.c -o src/ciphers/aes/aes_enc.o
+	$(LTCOMPILE) $(LTC_CFLAGS) $(CPPFLAGS) -DENCRYPT_ONLY -c src/ciphers/aes/aes.c -o src/ciphers/aes/aes_enc.o
 
 .c.o:
-	$(LTCOMPILE) $(LTC_CFLAGS) $(CPPFLAGS) $(LTC_LDFLAGS) -o $@ -c $<
+	$(LTCOMPILE) $(LTC_CFLAGS) $(CPPFLAGS) -o $@ -c $<
 
 LOBJECTS = $(OBJECTS:.o=.lo)
 
diff --git a/vcpkg-custom/triplets/fb-arm64-osx.cmake b/vcpkg-custom/triplets/fb-arm64-osx.cmake
index aa4d2e7484..fd8c1ca210 100644
--- a/vcpkg-custom/triplets/fb-arm64-osx.cmake
+++ b/vcpkg-custom/triplets/fb-arm64-osx.cmake
@@ -1,3 +1,5 @@
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_TARGET_ARCHITECTURE arm64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
@@ -8,3 +10,5 @@ set(VCPKG_OSX_ARCHITECTURES arm64)
 if(PORT STREQUAL "icu")
     set(VCPKG_LIBRARY_LINKAGE dynamic)
 endif()
+
+set(VCPKG_LINKER_FLAGS "-Wl,-headerpad_max_install_names" CACHE STRING "" FORCE)
diff --git a/vcpkg-custom/triplets/fb-x64-osx.cmake b/vcpkg-custom/triplets/fb-x64-osx.cmake
index 278f144408..b52a30ffc6 100644
--- a/vcpkg-custom/triplets/fb-x64-osx.cmake
+++ b/vcpkg-custom/triplets/fb-x64-osx.cmake
@@ -1,3 +1,5 @@
+set(VCPKG_BUILD_TYPE release)
+
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
 set(VCPKG_LIBRARY_LINKAGE static)
@@ -8,3 +10,5 @@ set(VCPKG_OSX_ARCHITECTURES x86_64)
 if(PORT STREQUAL "icu")
     set(VCPKG_LIBRARY_LINKAGE dynamic)
 endif()
+
+set(VCPKG_LINKER_FLAGS "-Wl,-headerpad_max_install_names" CACHE STRING "" FORCE)
